sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidRexExecutor.java

```diff
@@ -83,7 +83,11 @@ public void reduce(
         final RexNode literal;
 
         if (sqlTypeName == SqlTypeName.BOOLEAN) {
-          literal = rexBuilder.makeLiteral(exprResult.asBoolean(), constExp.getType(), true);
+          if (exprResult.valueOrDefault() == null) {
```

Comment:
This also seems similar to the earlier comment. Apart from array_contains and array_overlap this will also change some other results as well. Should we also think about backward compatibility for all of these or just say that these are the correct results now ?

Reply:
Does this need to check whether we're in 2VL mode and make a FALSE literal instead of NULL literal in that mode? Or in 2VL mode checked somewhere else?

Reply:
expressions don't really natively have a 2vl mode other than druid.generic.useDefaultValueForNull=false, so I suppose I could check `NullHandling.sqlCompatible()`

Reply:
this would be a behavior change though, since previously this would always return null, it just wasn't a `LONG` typed null, it was instead whatever the lhs argument was

Reply:
I wasn't suggesting we should change behavior, so let's keep it then.

Reply:
oh i guess the behavior change would be changing the native array_contains/array_overlap functions to not return null except in some cases.

This part of the code is actually a bug fix, since previously boolean constants weren't properly allowed to be null even though the native function might have been a null result, which I guess is a sort of behavior change though the previous behavior was certainly incorrect because the behavior of constant expressions that were considered boolean didn't match non-constant expressions.